File: src/chip8/timer.rs
pub struct Timer {
    value: u8,
}

impl Timer {
    pub fn new() -> Self {
        Self { value: 0 }
    }

    pub fn step(&mut self) {
        if self.value > 0 {
            self.value -= 1;
        }
    }

    pub fn read(&self) -> u8 {
        self.value
    }

    pub fn write(&mut self, value: u8) {
        self.value = value;
    }
}

File: chip8/timer.go
package chip8

type Timer struct {
	value uint8
}

func NewTimer() *Timer {
	return &Timer{}
}

func (t Timer) Read() uint8 {
	return t.value
}

func (t *Timer) Write(value uint8) {
	t.value = value
}

func (t *Timer) Step() {
	if t.value > 0 {
		t.value--
	}
}
